server/src/test/java/org/apache/druid/segment/metadata/SegmentMetadataQuerySegmentWalkerTest.java

```diff
@@ -0,0 +1,343 @@
+import org.apache.druid.query.QueryContexts;
+import org.apache.druid.query.QueryPlus;
+import org.apache.druid.query.QueryRunner;
+import org.apache.druid.query.QueryToolChest;
+import org.apache.druid.query.QueryToolChestWarehouse;
+import org.apache.druid.query.SegmentDescriptor;
+import org.apache.druid.query.TableDataSource;
+import org.apache.druid.query.context.ResponseContext;
+import org.apache.druid.query.metadata.SegmentMetadataQueryConfig;
+import org.apache.druid.query.metadata.SegmentMetadataQueryQueryToolChest;
+import org.apache.druid.query.metadata.metadata.AllColumnIncluderator;
+import org.apache.druid.query.metadata.metadata.SegmentAnalysis;
+import org.apache.druid.query.metadata.metadata.SegmentMetadataQuery;
+import org.apache.druid.query.spec.MultipleSpecificSegmentSpec;
+import org.apache.druid.server.coordination.ServerType;
+import org.apache.druid.server.initialization.ServerConfig;
+import org.apache.druid.server.metrics.NoopServiceEmitter;
+import org.apache.druid.timeline.DataSegment;
+import org.apache.druid.timeline.SegmentId;
+import org.apache.druid.timeline.VersionedIntervalTimeline;
+import org.apache.druid.timeline.partition.ShardSpec;
+import org.apache.druid.timeline.partition.SingleDimensionShardSpec;
+import org.easymock.EasyMock;
+import org.easymock.IAnswer;
+import org.joda.time.Interval;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mockito;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.EnumSet;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Random;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+public class SegmentMetadataQuerySegmentWalkerTest
+{
+  private final String DATASOURCE = "testDatasource";
+  private QueryToolChestWarehouse warehouse;
+  private DruidHttpClientConfig httpClientConfig;
+  private DruidServer[] servers;
+  private Random random;
+
+  @Before
+  public void setUp()
+  {
+    warehouse = new MapQueryToolChestWarehouse(
+        ImmutableMap.<Class<? extends Query>, QueryToolChest>builder()
+                    .put(
+                        SegmentMetadataQuery.class,
+                        new SegmentMetadataQueryQueryToolChest(
+                            new SegmentMetadataQueryConfig("P1W")
+
+                        )
+                    ).build());
+
+    httpClientConfig = new DruidHttpClientConfig()
+    {
+      @Override
+      public long getMaxQueuedBytes()
+      {
+        return 0L;
+      }
+    };
+
+    servers =
+        new DruidServer[]{
+            new DruidServer("test1", "test1", null, 10, ServerType.HISTORICAL, "bye", 0),
+            new DruidServer("test2", "test2", null, 10, ServerType.HISTORICAL, "bye", 0),
+            new DruidServer("test3", "test2", null, 10, ServerType.REALTIME, "bye", 0)
```

Comment:
I know this is just a test, but afaik nothing uses `REALTIME` anymore, so probably should use `INDEXER_EXECUTOR`...

`REALTIME` refers to the old realtime node type that was removed quite a while ago https://github.com/apache/druid/pull/7915

We should probably remove this `ServerType` completely now since it seems unlikely anyone will be rolling upgrade from such an old version at this point 😅